### ISSUE TITLE
Images not showing annotations (Missing in Forms template?)

### ISSUE THREAD

**Issue by pomeroyb** [Wikifab/wikifab-main#28]
I have been working to install WikiFab to use as internal documentation at our work, and it has solved a lot of problems for us! The only thing that's not working for me is the ImageAnnotator. I can open the annotator and add arrows and circles in the popup window:

![](https://i.imgur.com/XuDNRhf.png)

But when I click "Save", the annotations do no show up:

![image](https://user-images.githubusercontent.com/2469581/69017935-b5ce6680-095e-11ea-8f36-f8fd0fbf16b3.png)

I have error reporting turned on in LocalSettings.php:
`error_reporting( -1 );  

ini_set( 'display_errors', 1 );`

And I am not getting any errors. The annotations are being saved, because opening the annotator again will show the same circles and arrows being drawn.

When I init wikifab, I use the following:

`sudo php maintenance/initWikifab.php --setWikifabHomePage`

Which (I believe) uses the [english version of "Template_Tuto_StepImage.txt](https://github.com/Wikifab/wikifab-main/blob/master/maintenance/wikifabPages/en/Template_Tuto_StepImage.txt).

That file does not use the "#annotatedImage" tag described on the [Image Annotator Extension](https://github.com/Wikifab/mediawiki-extension-ImageAnnotator).

I have also tried using:
`sudo php maintenance/initWikifab.php --setWikifabHomePage --int`

Which (I believe) uses the [international version of "Template_Tuto_StepImage.txt](https://github.com/Wikifab/wikifab-main/blob/master/maintenance/wikifabPages/int/Template_Tuto_StepImage.txt). This file does use the "#annotatedImage" tag, but I still do not see the annotations when the site is switched to that.

Please let me know if there is anything I am missing, I would appreciate the help! :)

**Comment by pomeroyb** [comment]
Update to this:

I have reinstalled Wikifab using Method 1 from the README, using the 1.2.0-beta. Still no ImageAnnotator :(

It is now looking like the annotator is not creating the annotation thumbnails -- I see the `div class="annotatedcontent"` in the page source on a tutorial, but the `img class ="annotationlayer"` (That I see on the official WikiFab.org site) does not show up on my site.


I have also attempted to update the ImageAnnotator extension by extracting it [from Github](https://github.com/Wikifab/mediawiki-extension-ImageAnnotator) into my /extensions/ImageAnnotator/ folder, and running maintenance/update.php.

I did get some new behaviour: now whenever an image has an annotation, it shows the following:
![image](https://user-images.githubusercontent.com/2469581/69218252-3048e300-0b25-11ea-8d9a-3a418aebc50f.png)

That loading overlay never goes away, however.

Any ideas?

**Comment by pomeroyb** [comment]
Finally worked up the courage to find out this bug. This was tedious to find!



I installed a fresh install of Wikifab using Method 2 from the README. I first rolled back ImageAnnotator to version 1.1.0 (The same version, supposedly, that is installed on the wikifab.org website, [as seen on the Versions page](https://wikifab.org/w/index.php?title=Sp%C3%A9cial:Version)). _Side note: I believe that's another bug with the Packagist repo -- the version number never changes_

That worked for me, but it was pretty out of date. I then went forward, one commit at a time, to find the change that broke.

**The broken commit is "[Add API for annotated images](https://github.com/Wikifab/mediawiki-extension-ImageAnnotator/commit/92cde93cf9449e3014403bef25c80ea328f3d207)".**

If you are on the above branch, no annotated image is generated or shown, you just get the regular thumbnail of your image. A later commit (I didn't bother finding exactly which one) adds the loading overlay from the above post, but the problem persists.

To get the correct version when you run composer, first open composer.local.json:
```bash
$ cd /var/www/yourwebsite
$ nano composer.local.json
```
Change image-annotator to version 1.1.0 specifically:
```json
"wikifab/image-annotator": "dev-master#d0f2dc9ef873bb3763bef769ac55773d564285e3",
```

This is only a partial fix, however. It works for the Main Picture perfectly, but it appears there are still some issues when trying to display the annotated images in the steps. Will have to keep pushing on this. to make it work.

**Comment by pomeroyb** [comment]
Aha, finally tracked this down!

The issue is [line 250 of ApiImageAnnotatorThumb.php](https://github.com/Wikifab/mediawiki-extension-ImageAnnotator/blob/df89711bfa5607786e7611258585c67eb33f1465/includes/ApiImageAnnotatorThumb.php#L250)

```php
// convert to png :
$cmd = "inkscape -z -f ". escapeshellarg ($svgInFile) ." -w $width --export-background-opacity=0,0 --export-png=". escapeshellarg ($fileOut) ."";
```

I did not have inkscape installed on my distro 🤦‍♀ 

Running
`$ sudo apt install inkscape`

Solved this issue. Perhaps this requirement should be added to the Readme?
